enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,336 @@
+Specifically the [PlatformSpec](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L47)
+and [PlatformStatus](https://github.com/openshift/api/blob/51f399230d604fa013c2bb341040c4ad126e7309/config/v1/types_infrastructure.go#L65) parts
+of the infrastructure resource.
+
+Given that PlatformSpec and PlatformStatus are defined as "discriminated unions" and are required to have the platform type encoded within
+"openshift/api" beforehand, it creates a significant footprint and plenty of effort from Red Hat engineers to create the initial technical enablement of
+a new cloud-provider and mostly impossible without Red Hat engineering engagement.
+
+Since a lot of infrastructure related components (such as [CCCMO](https://github.com/openshift/cluster-cloud-controller-manager-operator), [Machine API operator](https://github.com/openshift/machine-api-operator), [Machine Config Operator](https://github.com/openshift/machine-config-operator), and so on)
+require information about, at least, the presence of an underlying cloud-provider, the "None" platform does not fit well as a signal in such situation.
+
+Having a special, built-in, and somewhat generic platform type which will signal about the presence of an underlying infrastructure platform without
+platform specific details will help to reduce the number of changes across OCP repositories
+and simplify the initial integration work for non-Red Hat contributors.
+
+In the future, at some degree, [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+API might help to solve the issue and give a room for supplementing *some* platform dependant components, however some integral parts of 
+OpenShift can not be disabled and still requires a signal about underlying platform, for example KCM and MCO in respect of enablement of
+[external cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/)
+
+Examples of difficulties which are present today due to having a predefined list of platforms:
+- No defined mechanism to set `--cloud-provider=external` arg to kubelet/KCM/apiserver without merges and further revendoring of "openshift/api"
+- No way to extend machine-api and deliver new provider without merges to "openshift/machine-api-operator" and "openshift/api" repos
+
+### User Stories
+
+- As a cloud-provider affiliated engineer / platform integrator / RH partner
+  I want to have a mechanism to signal Openshift's built-in operators about additional
+  cloud-provider specific components which should be delivered and managed by its own or manually.
+  
+  We are aware about examples of supplementing OKD installations with a custom machine-api controllers,
+  however users are experiencing a lot of difficulties on this path due to necessity of, literally, reverse engineering,
+  manual management of generic MAPI controllers and so on.
```

Comment:
imo, we should give an example through the user stories. i think the first story is good but we should have something with specifics so that other readers know what we are talking about.

for example,

As a cloud provider whose platform is not integrated into OpenShift, I want to have the Cloud Controller Manager for my infrastructure running in OpenShift from the initial install. Having a platform type that allows for the addition of operators which perform platform-specific functionality would help me to create better integrations between OpenShift and my infrastructure platform.